### PR TITLE
Update eigen to v3.4-rc1 (3.3.91)

### LINKED ISSUE #6833: Test Eigen 3.4-rc1
We should check if CMSSW and the externals are happy with 3.4-rc1, and report any problems:

> Eigen 3.4-rc1 has been released on April 19, 2021.
> 
> Depending on the amount of reported issues, 3.4 will be released late April or early May.
> 
> The source archive is available at: https://gitlab.com/libeigen/eigen/-/releases/3.4-rc1
> Please check the [release page](https://eigen.tuxfamily.org/index.php?title=3.4) for a list of changes.

Can we try building it for the various compilers and architectures of the `master` branch ?

### PR DESCRIPTION
**Pull request by mrodozov**
Resolves: #6833

### CODE REVIEW

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
please test for slc7_aarch64_gcc9

**Comment by mrodozov** [comment]
please test for slc7_ppc64le_gcc9

**Comment by mrodozov** [comment]
@fwyzard TF 2.4.1 disagrees with new eigen
the 2.5.0-rc1 (latest tag) tensorflow is having this source for eigen:
https://github.com/tensorflow/tensorflow/blob/v2.5.0-rc1/third_party/eigen3/workspace.bzl#L9

-> https://gitlab.com/libeigen/eigen/-/commit/f612df273689a19d25b45ca4f8269463207c4fee

which is between 3.3 and 3.4 (month ago old) and new eigen might work with TF 2.5.0-rc1

**Comment by fwyzard** [comment]
> @fwyzard TF 2.4.1 disagrees with new eigen
> the 2.5.0-rc1 (latest tag) tensorflow is having this source for eigen:
> https://github.com/tensorflow/tensorflow/blob/v2.5.0-rc1/third_party/eigen3/workspace.bzl#L9
> 
> -> https://gitlab.com/libeigen/eigen/-/commit/f612df273689a19d25b45ca4f8269463207c4fee
> 
> which is between 3.3 and 3.4 (month ago old) and new eigen might work with TF 2.5.0-rc1

Is there a way to force the build to continue even if TF ans what depend on it fail ?
We could check at least if the code in CMSSW that depends explicitly on Eigen builds or not.

File: eigen.spec [hunk 1]
```diff
@@ -1,7 +1,7 @@
-### RPM external eigen 011e0db31d1bed8b7f73662be6d57d9f30fa457a
+### RPM external eigen 831133c
 ## INITENV +PATH PKG_CONFIG_PATH %{i}/share/pkgconfig
 ## NOCOMPILER
-%define tag fd0c254c0a9adc47c094257ce6256ab4d94b956e
+%define tag 76141596f2a3f06b1659d00dc4a4dcccd8750a7d
```
Comment (fwyzard) [inline eigen.spec:4]: Did we already try to submit upstream the two changes we apply locally o the Eigen sources ?

**Comment by mrodozov** [comment]
> Is there a way to force the build to continue e

I think those are the externals that depend on eigen:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a2297d/14591/external/

File: eigen.spec (hunk 1 above)
Comment (mrodozov) [reply eigen.spec:4]: not as far as I know - I didn't do it and not sure if our changes are generic enough to propose them

**Comment by smuzaffar** [comment]
closing this, open a new one with TF 2.5.0

**Pull request closed without merging**